akka-stream/src/main/scala/akka/stream/impl/io/TcpConnectionStream.scala

```diff
@@ -91,7 +89,6 @@ private[akka] abstract class TcpStreamActor(val settings: ActorMaterializerSetti
       if (!closed) {
         closed = true
         pendingElement = null
-        if (!tcpOutputs.isFlushed && (connection ne null)) connection ! ResumeReading
```

Comment:
@drewhk, I thought about this a little bit. So here is short list:
1. One line was removed that is sending ResumeReading to Connection, causing next read cycle.
2. Method is doing closed=true so this is changing state (of TcpConnection) to drain-and-discard mode I believe
3. This change is not causing switching to another Receive (like actor.become(newReceive)) so we should not miss any close events (in TcpStreamActor that we get fromTcp Connection)
4. The only case I can imagine is that we are closing TcpStreamActor from stream side with expectation that will trigger reading from TcpConnection
5. this is the only place in tcpInputs where we care about tcpOutputs (isFlushed). So this sounds like if input is closing on Steam side and for some reason we haven't sent all data to TcpConnection - we need to Resume Reading. Sounds weired.

Anyway, if you can give steps to reproduce I will create unit test and will go from there